src/modules/item/sagas.ts

```diff
@@ -142,9 +143,16 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
   }
 
   function* handleFetchItemsRequest(action: FetchItemsRequestAction) {
-    const { address } = action.payload
+    const { address, itemIds } = action.payload
     try {
-      const items: Item[] = yield call(() => legacyBuilder.fetchItems(address))
+      let items: Item[]
+
+      if (itemIds) {
+        items = yield call([Promise, 'all'], [itemIds.map(id => legacyBuilder.fetchItem(id))])
```

Comment:
Since this is being triggered per collection, couldn't we use the `fetchCollectionItems` so we avoid doing one request per each item of the collection? 
Instead of receiving an array of `itemIds` we could expect a `collectionId` and fetch all the items at once.

Reply:
You are right. I added that before realizing there was a `fetchCollectionItems` action. Changing it.

Reply:
Offtopic: `yield all([])` also works (instead of using a call on Promise.all)

Reply:
Nice tip

Reply:
Ended up removing the `itemIds` as they were unnecessary.